.github/workflows/build-project.yaml

```diff
@@ -43,17 +52,73 @@ jobs:
+        with:
+          java-version: "${{ matrix.java }}"
+      - uses: actions/cache@v2
+        with:
+          path: "${{ github.workspace}}/libraries"
+          key: "${{runner.os}}-${{matrix.java}}-it-libraries"
+      - name: Download SpongeVanilla server
+        uses: actions/download-artifact@v2
+        with:
+          name: SpongeVanilla Production Jar
+      - name: Run SpongeVanilla Test (windows)
+        if: "runner.os == 'Windows'"
+        run: java -jar $(gci | Where-Object NameString -Match "-universal.jar") --launchTarget sponge_server_it
+      - name: Run SpongeVanilla Test (other)
+        if: "runner.os != 'Windows'"
+        run: java -jar *-universal.jar --launchTarget sponge_server_it
+  publish:
+    if: "${{ github.event_name == 'push' && github.ref == 'refs/heads/api-8' }}"
+    needs: [build, integrationTest]
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          submodules: recursive
+      - uses: actions/cache@v2
+        with:
+          path: |
+            ~/.gradle/caches
+            ~/.gradle/wrapper
+            ${{ github.workspace}}/buildSrc/.gradle/
+            ${{ github.workspace}}/forge/build/fg_cache
+            ${{ github.workspace}}/.gradle/
+          key: "${{ runner.os }}-minecraft-${{ hashFiles('**/*.gradle*') }}"
+          restore-keys: |
+            ${{ runner.os }}-minecraft-
+      - name: Set up JDK 1.8
+        uses: actions/setup-java@v1
+        with:
+          java-version: 1.8
+        # We don't need to run tests again, so we just publish
+      - name: Setup workspace
+        run: |
+          echo "GIT_COMMIT=${GITHUB_SHA}" >> $GITHUB_ENV
+          echo "GIT_BRANCH=${GITHUB_REF##*/}" >> $GITHUB_ENV
+          echo "BUILD_NUMBER=${GITHUB_RUN_NUMBER}" >> $GITHUB_ENV
       - name: Publish to Sponge Maven & GitHub Packages
-        if: github.events.push
-        run: ./gradlew publish
+        run: ./gradlew -s :publish :SpongeVanilla:publish
         env:
-          GITHUB_USERNAME: ${{ github.actor }}
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-          ORG_GRADLE_PROJECT_spongeSnapshotRepo: ${{ secrets.SPONGE_MAVEN_SNAPSHOT_REPO_URL }}
-          ORG_GRADLE_PROJECT_spongeReleaseRepo: ${{ secrets.SPONGE_MAVEN_RELEASE_REPO_URL }}
-          ORG_GRADLE_PROJECT_spongeUsername: ${{ secrets.SPONGE_MAVEN_REPO_USER }}
-          ORG_GRADLE_PROJECT_spongePassword: ${{ secrets.SPONGE_MAVEN_REPO_PASSWORD }}
-
-
+          CI_SYSTEM: Github Actions
+          GITHUB_USERNAME: "${{ github.actor }}"
+          GITHUB_TOKEN: "${{ secrets.GITHUB_TOKEN }}"
+          ORG_GRADLE_PROJECT_spongeSnapshotRepo: "${{ secrets.SPONGE_MAVEN_SNAPSHOT_REPO_URL }}"
+          ORG_GRADLE_PROJECT_spongeReleaseRepo: "${{ secrets.SPONGE_MAVEN_RELEASE_REPO_URL }}"
+          ORG_GRADLE_PROJECT_spongeUsername: "${{ secrets.SPONGE_MAVEN_REPO_USER }}"
+          ORG_GRADLE_PROJECT_spongePassword: "${{ secrets.SPONGE_MAVEN_REPO_PASSWORD }}"
```

Comment:
These shouldn't be double quoted (or rather, there's no obvious reason why these are being double quoted)